tests/console/python3_setuptools.pm

```diff
@@ -19,64 +20,80 @@ use serial_terminal 'select_serial_terminal';
-    my @package_loc = split(/:/, script_output("pip$python_spec_release[1] show user_package_setuptools | grep Location"));
+}
+
+# Installs the package from local archive and verify it using the sample Python module 'test_module.py'.
+sub local_install_test ($version_number) {
+    validate_script_output("pip$version_number install dist/user_package_setuptools-1.0.tar.gz", sub { m/Successfully installed/ });
+    my @package_loc = split(/:/, script_output("pip$version_number show user_package_setuptools | grep Location"));
     validate_script_output("ls $package_loc[1]", sub { m/user_package_setuptools/ });
-    assert_script_run("python$python_spec_release[1] sample_test_module/test_module.py");
+    assert_script_run("python$version_number sample_test_module/test_module.py");
+    uninstall_package($version_number);
+}
+
+# Installs the package from http server and verify it using the sample Python module 'test_module.py'.
+sub http_install_test ($version_number) {
+    # prepare repository structure.
+    # Notice the directory must be in normalized form (https://packaging.python.org/en/latest/specifications/name-normalization/#name-normalization)
+    assert_script_run "mkdir -p repo_webroot/user-package-setuptools && cp dist/*.tar.gz repo_webroot/user-package-setuptools && pushd repo_webroot";
+    assert_script_run "pip$version_number install wheel";
+    # spin up a local http server just for this installation
+    my $server_pid = background_script_run "python$version_number -m http.server";
+    # install the package from http server
+    assert_script_run "pip$version_number install --no-build-isolation -i http://localhost:8000 user_package_setuptools";
+    # close server and restore directory
+    assert_script_run "kill $server_pid && popd";
+    assert_script_run("python$version_number sample_test_module/test_module.py");
+    uninstall_package($version_number);
 }
 
 # Uninstall the installed user_package_setuptools and verify it.
-sub cleanup_package {
-    my ($python3_release_version) = @_;
-    my $python_version = get_python3_binary($python3_release_version);
-    my @python_spec_release = split("python", $python_version);
-    assert_script_run("pip$python_spec_release[1] uninstall user_package_setuptools -y");
-    my $out = script_output "$python_version sample_test_module/test_module.py", proceed_on_failure => 1;
+sub uninstall_package ($version_number) {
+    assert_script_run("pip$version_number uninstall user_package_setuptools -y");
+    my $out = script_output "python$version_number sample_test_module/test_module.py", proceed_on_failure => 1;
     die("pip uninstall failed for the package") if (index($out, "ModuleNotFoundError") == -1);
-    # Deletion of user_package_setuptools.egg-info and dist folder
-    assert_script_run("rm -r user_package_setuptools.egg-info");
-    assert_script_run("rm -r dist");
-    assert_script_run("deactivate");
+}
 
+sub cleanup {
```

Comment:
it would be good to have this called on post_run and post_fail, and just to have an extra note, add a comment in the `deactivate` step of the virtualenv